tardis/rest/app/routers/resources.py

```diff
@@ -10,20 +11,32 @@
 async def get_resource_state(
     drone_uuid: str = Path(..., regex=r"^\S+-[A-Fa-f0-9]{10}$"),
     sql_registry: SqliteRegistry = Depends(database.get_sql_registry()),
-    _: str = Security(security.check_authorization, scopes=["resources:get"]),
+    _: AuthJWT = Security(security.check_authorization, scopes=[Resources.get]),
 ):
     query_result = await crud.get_resource_state(sql_registry, drone_uuid)
     try:
         query_result = query_result[0]
     except IndexError:
-        raise HTTPException(status_code=404, detail="Drone not found") from None
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Drone not found"
+        ) from None
     return query_result
 
 
 @router.get("/", description="Get list of managed resources")
 async def get_resources(
     sql_registry: SqliteRegistry = Depends(database.get_sql_registry()),
-    _: str = Security(security.check_authorization, scopes=["resources:get"]),
+    _: AuthJWT = Security(security.check_authorization, scopes=[Resources.get]),
 ):
     query_result = await crud.get_resources(sql_registry)
     return query_result
+
+
+@router.patch("/{drone_uuid}/drain", description="Gently shut shown drone")
+async def drain_drone(
+    drone_uuid: str = Path(..., regex=r"^\S+-[A-Fa-f0-9]{10}$"),
+    sql_registry: SqliteRegistry = Depends(database.get_sql_registry()),
+    _: AuthJWT = Security(security.check_authorization, scopes=[Resources.patch]),
+):
+    await crud.set_state_to_draining(sql_registry, drone_uuid)
```

Comment:
Is this enough to drain a drone? The status in the DB is changed only. @giffels Is there an additional function call of the batch system adapter necessary, such as in [htcondor batch system adapter](https://github.com/MatterMiners/tardis/blob/6ae4b9385d4eb40ec697e700d900d251861f031b/tardis/adapters/batchsystems/htcondor.py#L117)?

Reply:
Yes, in combination with #260 it is.

Reply:
Okay, thanks for the explanation. :-)